test/task_cancel_test.cpp

```diff
@@ -51,11 +51,33 @@ task<int> bar() {
   co_return -1;
 }
 
+task<inplace_stop_token> get_token_inner() {
+  co_return co_await get_stop_token;
+}
+
+task<inplace_stop_token> get_token_outer() {
+  auto a = co_await get_stop_token;
+  auto b = co_await get_token_inner();
+  EXPECT_EQ(a, b);
+  co_return b;
+}
+
 task<void> void_test() {
   co_await stop;
```

Comment:
Looks like you've got a local `stop` defined; perhaps you should delete it so that this is using the thing you've added.

Also, I think `co_await stop;` and `co_await stop_if_requested;` look _weird_; it would be way more normal-looking to my eyes if you `co_await`ed the result of calling them: `co_await stop();` and `co_await stop_if_requested();`.  Did you have a reason for making them values instead of callables?

Reply:
> Did you have a reason for making them values instead of callables?

Slightly less typing for me and everybody else? That's it, really; no other reason.

Reply:
It occurs to me that maybe I should get over the feeling that `co_await <value>;` is weird: it's the fact that it's being `co_await`ed that matters so the weirdness makes a different thing look different.  This might be important.  It feels related to "`std::move` doesn't move"; if it were `co_await stop();` then we'd have to say "`stop()` doesn't stop".  I suspect that no one would expect
```
task<void> foo() {
  stop;
}
```
to do anything, but they might expect
```
task<void> foo() {
  stop();
}
```
to do something.